src/transformStateWithClones.js

```diff
@@ -6,8 +6,45 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const arrayWithNewStates = [];
+
+  for (const action of actions) {
+    transform(newState, action);
+
+    const copyNewState = { ...newState };
+
+    arrayWithNewStates.push(copyNewState);
```

Comment:
seems like a variable copyNewState is redundant 
```suggestion
    transform(newState, action);
    arrayWithNewStates.push(...newState);
```